compiler/rustc_typeck/src/check/expr.rs

```diff
@@ -2058,8 +2058,8 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
     ) -> Option<(&Vec<ty::FieldDef>, SubstsRef<'tcx>)> {
         debug!("get_field_candidates(span: {:?}, base_t: {:?}", span, base_t);
 
-        let mut autoderef = self.autoderef(span, base_t);
-        while let Some((base_t, _)) = autoderef.next() {
+        let autoderef = self.autoderef(span, base_t);
+        for (base_t, _) in autoderef {
```

Comment:
You don't need a temporary here I think.
```suggestion
        for (base_t, _) in self.autoderef(span, base_t) {
```